app/default_overrides.go

```diff
@@ -149,6 +151,21 @@ func (ibcModule) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
 	return cdc.MustMarshalJSON(gs)
 }
 
+// icaModule defines a custom wrapper around the ica module to provide custom
+// default genesis state.
+type icaModule struct {
+	ica.AppModuleBasic
+}
+
+// DefaultGenesis returns custom ica module genesis state.
+func (icaModule) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
+	gs := icagenesistypes.DefaultGenesis()
+	gs.HostGenesisState.Params.AllowMessages = icaAllowMessages()
+	gs.HostGenesisState.Params.HostEnabled = true
+	gs.ControllerGenesisState.Params.ControllerEnabled = false
+	return cdc.MustMarshalJSON(gs)
+}
+
```

Comment:
[No action needed] It's interesting that we've had to develop this pattern of wrapping the modules. It makes me think that the default genesis should be provided in the constructor

Reply:
TBH I also thought this pattern was a bit strange but I didn't see another way to override default genesis params.